verilog/tools/ls/symbol-table-handler.cc

```diff
@@ -248,7 +251,43 @@ std::vector<verible::lsp::Location> SymbolTableHandler::FindDefinitionLocation(
   const verible::TextStructureView &text = parsedbuffer->parser().Data();
 
   const verible::TokenInfo cursor_token = text.FindTokenAt(cursor);
-  auto symbol = cursor_token.text();
+  return cursor_token.text();
+}
+
+std::optional<verible::lsp::Location>
+SymbolTableHandler::GetLocationFromSymbolTableNode(
+    absl::string_view symbol, const VerilogSourceFile *file_origin) {
+  // TODO (glatosinski) add iterating over multiple definitions
+  verible::lsp::Location location;
+  if (!file_origin && curr_project_) {
+    file_origin = curr_project_->LookupFileOrigin(symbol);
+  }
+  if (!file_origin) {
+    LOG(ERROR) << "Origin file not available";
+    return {};
+  }
+  location.uri = PathToLSPUri(file_origin->ResolvedPath());
+  auto *textstructure = file_origin->GetTextStructure();
+  if (!textstructure) {
+    LOG(ERROR) << "Origin file's text structure is not parsed";
+    return {};
+  }
+  verible::LineColumnRange symbollocation =
+      textstructure->GetRangeForText(symbol);
```

Comment:
That symbol_name getting from the text structure must be coming from that text structure before (because it must be a substring), otherwise you can not look it up.

Do we know that is the case ?
And if so: why do we have to check that GetTextStructure() returned an object as we must've gotten the symbol from there.